packages/asset-swapper/src/utils/market_operation_utils/fills.ts

```diff
@@ -119,18 +119,15 @@ function dexQuotesToPaths(
     fees: { [source: string]: BigNumber },
 ): Fill[][] {
     const paths: Fill[][] = [];
-    for (const quote of dexQuotes) {
+    for (let quote of dexQuotes) {
         const path: Fill[] = [];
+        // Drop any non-zero entries. This can occur if the
+        // first few fills on Kyber were UniswapReserves
+        quote = quote.filter(q => !q.output.isZero());
```

Comment:
I thought this too originally but didn't seem to encounter it. 

Let's say the samples for Kyber looks like: `[0, 0, 1000]` Where `0`s are other reserves, we only unlock kyber at the third stop (enough volume for it to switch). In the routing we'd only select Kyber for at least this amount, so we're going to hit the right reserve. But that could all change at execution time.

I expected to see it in simbot, but it didn't seem to occur, so I removed sorting Kyber last.

Reply:
If we can gather compelling data on pairs where this scenario is most likely then I'm fine with this hack. We should def document this reasoning and accepted risk in the code though. Otherwise I'll forget one day and revert this change.